src/rust/engine/src/externs/mod.rs

```diff
@@ -211,14 +211,18 @@ pub fn getattr_from_str_frozendict<'p, T: FromPyObject<'p>>(
     .collect()
 }
 
-pub fn getattr_as_optional_string(value: &PyAny, field: &str) -> Option<String> {
-  let v = value.getattr(field).unwrap();
-  if v.is_none() {
-    return None;
+pub fn getattr_as_optional_string(value: &PyAny, field: &str) -> PyResult<Option<String>> {
```

Comment:
@stuhood Something more like this?

I haven't delved deep enough into this workings of this to know for sure how `PyResult<None>` will be handled. 

Thank you for the insight!

Reply:
That signature, yea: but then moving the body of the function back to how it original was, and using `?` rather than `unwrap()`: i.e.:
```rust
let v = value.getattr(field)?;
```

Reply:
Would it be preferred to add a `pyo3` dependency to the `fs` crate? That would permit conversion of a PyErr to the `StoreError` and thus use of the `?` in functions like the following:

https://github.com/pantsbuild/pants/blob/main/src/rust/engine/src/nodes.rs#L336-L339

Otherwise it'd be a tad more verbose with something like:
```
    let working_directory = externs::getattr_as_optional_string(value, "working_directory")
        .map_or_else(|none| None, |dir| {
        if let Some(d) = dir {
            RelativePath::new(d).ok()
        } else {
            None
        }
    });
```